### PR TITLE
Feature/#155 fix erratic ticks behaviour

### PR DESCRIPTION
**Pull request by mridulmeh**
**Before submitting a pull request,** please make sure the following is done:

1. Fork [the repository](https://github.com/chartshq/muze).
2. Initialize `git flow` and create a separate branch from develop.
3. Run `npm install` in the repository root.
4. Make sure your code lints (`npm run lint`)

### CODE REVIEW

File: packages/layout/src/visual-matrix/column-matrix.js [hunk 1]
```diff
@@ -143,11 +143,7 @@ export default class ColumnVisualMatrix extends VisualMatrix {
             heightDist = heightDist.map(() => 0);
         } else {
             conditions.forEach((e) => {
-                if (remainaingHeight / divider < maxHeights[e]) {
-                    heightDist[e] = 0;
-                } else {
-                    heightDist[e] = Math.min(maxHeights[e], (remainaingHeight) / divider);
-                }
+                heightDist[e] = Math.min(maxHeights[e], (remainaingHeight) / divider);
```
Comment (UD-UD) [inline packages/layout/src/visual-matrix/column-matrix.js]: use meaning full variable. not `e`

File: packages/muze-axis/src/cartesian-axis/continous-axis.js [hunk 2]
```diff
@@ -158,8 +159,21 @@ export default class ContinousAxis extends SimpleAxis {
...
             return this;
         }
-        axis.tickValues(this.getTickValues());
+        const newTickValues = this.getTickValues();
+
+        axis.tickValues(newTickValues);
+        const smartLabel = this.dependencies().labelManager;
+        smartLabel.setStyle(this._tickLabelStyle);
 
+        const smartTicks = newTickValues.map((e) => {
```
Comment (UD-UD) [inline packages/muze-axis/src/cartesian-axis/continous-axis.js]: use meaningfull variable

File: packages/muze-axis/src/cartesian-axis/helper.js [hunk 3]
```diff
@@ -42,8 +42,13 @@ export const getNumberOfTicks = (availableSpace, labelDim, axis, axisInstance) =
     }
 
     numberOfValues = Math.min(numberOfTicks, Math.max(2, numberOfValues));
+    let tickValues = axis.scale().ticks(numberOfValues);
 
-    return axis.scale().ticks(numberOfValues);
+    if (numberOfValues === 2) {
```
Comment (UD-UD) [inline packages/muze-axis/src/cartesian-axis/helper.js]: `2` should be defined in a constant. and That constant should be used. Enable more readablity.

File: packages/muze-axis/src/cartesian-axis/helper.js [hunk 4]
```diff
@@ -42,8 +42,13 @@ export const getNumberOfTicks = (availableSpace, labelDim, axis, axisInstance) =
...
+        tickValues = axis.scale().ticks(10);
```
Comment (UD-UD) [inline packages/muze-axis/src/cartesian-axis/helper.js]: Same as above

**Review by UD-UD — CHANGES_REQUESTED** [review]
In some files its seen that mathematical expressions are use in-line if statement, If the mathematical expression is is long and calculates meaning full values use a meaningful variable to store the result of that expression and then use the same in required places.
For eg expressions like this `availHeight - top - bottom - tickShifter`

**Comment by mridulmeh** [comment]
In most of the cases it has been avoided, but if has been, it's probably because the need of the calculation only exists in the if else statement. So, I did not feel the need to create a variable. However, this can be taken into account in separate refactoring issue.

**Review by UD-UD — APPROVED** [review]

**Pull request closed without merging**